ga4gh/datamodel/ontologies.py

```diff
@@ -0,0 +1,82 @@
+"""
+Support for Ontologies.
+FIXME: this is currently hard-coded for the GENCODE set as a short-term hack
+to support BRCA changeling.
+"""
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import ga4gh.protocol as protocol
+
+
+class OntologyTermSet(object):
+    """A set of related ontology terms.
+    """
+    def __init__(self, ontologySource):
+        self.ontologySource = ontologySource
+        self.byId = dict()
+        self.byName = dict()
+
+    def add(self, ontologyTerm):
+        "add an ontology term to the object"
+        if ontologyTerm.id in self.byId:
+            raise ValueError(
+                "OntologyTerm id already exists: "
+                + str(ontologyTerm))
+        self.byId[ontologyTerm.id] = ontologyTerm
+        if ontologyTerm.name is not None:
+            if ontologyTerm.name in self.byName:
+                raise ValueError(
+                    "OntologyTerm idan already exists: "
+                    + str(ontologyTerm))
+
+    def create(self, id, name):
+        "create a new ontology term"
+        self.add(OntologyTerm(self.ontologySource, id, name))
+
+
+class OntologyTerm(object):
+    """A specific ontology term."""
+    def __init__(self, ontologySource, id, name):
+        self.__ontologySource = ontologySource
+        self.__id = id
+        self.__name = name
+
+    def __str__(self):
+        return self.ontologySource + "/" + self.id + "/" + str(self.name)
```

Comment:
string formatting should be done with the `.format` method ... both as a matter of style and because the way you're doing it is slightly less efficient in creating some strings that are immediately discarded